testvectors/testvectors.go

```diff
@@ -0,0 +1,60 @@
+package testvectors
+
+import (
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+
+	registry "github.com/oasisprotocol/metadata-registry-tools"
+)
+
+const keySeedPrefix = "oasis-metadata-registry test vectors: "
+
+// EntityMetadataTestVector is an entity metadata test vector.
+type EntityMetadataTestVector struct {
+	Kind                    string                        `json:"kind"`
+	SignatureContext        string                        `json:"signature_context"`
+	EntityMeta              registry.EntityMetadata       `json:"descriptor"`
+	SignedEntityMeta        registry.SignedEntityMetadata `json:"signed_descriptor"`
+	EncodedEntityMeta       []byte                        `json:"encoded_descriptor"`
+	EncodedSignedEntityMeta []byte                        `json:"encoded_signed_descriptor"`
+	Valid                   bool                          `json:"valid"`
+	SignerPrivateKey        []byte                        `json:"signer_private_key"`
+	SignerPublicKey         signature.PublicKey           `json:"signer_public_key"`
+}
+
+// MakeEntityMetadataTestVector generates a new test vector from an entity metadata.
+func MakeEntityMetadataTestVector(kind string, meta *registry.EntityMetadata, valid bool) EntityMetadataTestVector {
+	signer := memorySigner.NewTestSigner(keySeedPrefix + kind)
+	return MakeEntityMetadataTestVectorWithSigner(kind, meta, valid, signer)
+}
+
+// MakeEntityMetadataTestVectorWithSigner generates a new test vector from an entity metadata using a specific signer.
+func MakeEntityMetadataTestVectorWithSigner(
+	kind string,
+	meta *registry.EntityMetadata,
+	valid bool,
+	signer signature.Signer,
+) EntityMetadataTestVector {
+	sigMeta, err := registry.SignEntityMetadata(signer, meta)
+	if err != nil {
+		panic(err)
+	}
+
+	sigCtx, err := signature.PrepareSignerContext(registry.EntityMetadataSignatureContext)
+	if err != nil {
+		panic(err)
+	}
+
+	return EntityMetadataTestVector{
+		Kind:                    kind,
+		SignatureContext:        string(sigCtx),
+		EntityMeta:              *meta,
+		SignedEntityMeta:        *sigMeta,
+		EncodedEntityMeta:       cbor.Marshal(meta),
+		EncodedSignedEntityMeta: cbor.Marshal(sigMeta),
+		Valid:                   valid,
```

Comment:
Doesn't the `Valid` field in the `*TestVector` struct refer to "is this a valid signature", which i believe should be `True` even for signatures of "invalid metadata registry content" (which i think is kinda irrelevant for the test-vectors as that's just a registry limit, and the signer should probably handle correctly both).

Reply:
I discussed this with @kostko and the `valid` field for the [Oasis Core's Transaction Test Vectors](https://docs.oasis.dev/oasis-core/high-level-components/index/test-vectors) is defined quite broadly :slightly_smiling_face: :

> `valid` is a boolean flag indicating whether the given test vector represents a valid transaction.

That being said, I agree that checking something like the validity of an URL, email, Keybase or Twitter scheme would be an overkill for a HSM signer app.

I'll split the test cases into 2, ones that are simple size checks and the other ones that check the semantics of a field's value, and update `gen_vectors` to only create test vectors with the first set.

Reply:
@kostko, @ptrus, I've updated the PR, plese take another look.

Reply:
Yeah I think this makes more sense now (test vectors not including all of the semantic checks). Thanks.

Reply:
> Yeah I think this makes more sense now (test vectors not including all of the semantic checks). Thanks.

No problem. Thanks for noticing and flagging this yesterday!